include/swift/AST/Types.h

```diff
@@ -2397,7 +2399,6 @@ getSILFunctionLanguage(SILFunctionTypeRepresentation rep) {
 class AnyFunctionType : public TypeBase {
   const Type Input;
```

Comment:
If we store the parameters directly why do we store the input type as well?

Reply:
The subtypes use `llvm::TrailingObjects` to store `AnyFunctionType::Param` values, and therefore they need to track the number of input parameters.

Reply:
That is a good question. I imagine that in practice that `const Type Input` is effectively a cache. In any case, that is beyond the scope of this PR.

Reply:
Mea culpa: Haven't finished migrating clients away from the old representation.  We shunt back and forth between the two still - would require more work than just one or two PRs to switch.

Reply:
So I've been poking around this area. @codafi – Where in your priority list is eliminating `getInput()`?

Reply:
Somewhere at or below eliminating `InOutType`.  That's a much more feasible goal than re-formulating the way the type checker looks at paren types.